content/en/docs/tasks/administer-cluster/migrating-from-dockershim/ migrating-telemetry-and-security-agents.md

```diff
@@ -0,0 +1,99 @@
+
+### Why do telemetry agents talk to Docker?
+
+Historically, kubernetes was built on top of Docker. Kubernetes is managing networking and scheduling, Docker was placing and operating containers on a node. So you can get scheduling-related metadata like a pod name from Kubernetes and containers state information from Docker. Over time more runtimes were created to manage containers. Also there are projects and kubernetes features that generalize container status information extraction across many runtimes.
+
+But some agents are still Docker-dependent. They may run commands like [docker
+ps](https://docs.docker.com/engine/reference/commandline/ps/) or [docker
+top](https://docs.docker.com/engine/reference/commandline/top/) to list
+containers and processes or [docker
+logs](https://docs.docker.com/engine/reference/commandline/logs/) to subscribe
+on docker logs. With the deprecating of Docker as a container runtime, these
+commands will not work any longer.
+
+
+### Detecting Docker dependency from the agents-daemonsets
+
+If a pod wants to make calls to the `dockerd` running on the node, the pod must mount the docker daemon as a volume.
+
+For example, On COS images, docker exposes it’s unix socket at
+`/var/run/docker.sock` This means that the customer’s pod spec will include a
+`hostPath` volume mount of `/var/run/docker.sock`.
+
+The script to find all pods mounting this volume may look like this:
+
+```bash
+kubectl get pods --all-namespaces \
+-o=jsonpath='{range .items[*]}{"\n"}{.metadata.namespace}{":\t"}{.metadata.name}{":\t"}{range .spec.volumes[*]}{.hostPath.path}{", "}{end}{end}' \
+| sort \
+| grep '/var/run/docker.sock'
+```
+
+(this script outputs the namespace and name of the pod. You can remove the grep `/var/run/docker.sock` to review other mounts)
+
+Note, there are alternative ways to call a Docker. For instance, the parent
+directory `/var/run` may be mounted instead of the full path (like in [this
+example](https://gist.github.com/itaysk/7bc3e56d69c4d72a549286d98fd557dd)). So
+the script above will only detect the most common uses.
+
+
+### Detecting Docker dependency from node deployed agents
+
+In case your cluster nodes are customized and install additional security and
+telemetry agents on the node, make sure to check with the vendor of the agent whether it has dependency on Docker.
+
+## Telemetry and security agent vendors
+
+This section is intended to collect information about various telemetry and
+security agents that may have dependency on containers runtime.
+
+This section is work in progress.
+
+{{< note >}}
+
+Vendors, please help to fill out this section. See
+[CONTRIBUTING](https://github.com/kubernetes/website/blob/master/CONTRIBUTING.md)
+guide for instructions. Or create an issue if you need help formatting the
+document.
+
+{{< /note >}}
```

Comment:
I would omit this, replacing it with a draft pull request once the other documentation has landed. We don't publish work-in-progress documentation.

Alternatively, hyperlink to a wiki page run by SIG Node, or something similar. Linking to incomplete content is OK.

Reply:
Would the list of vendors who likely has dependency on Docker be OK to have alongside with this request? Let me draft it...